src/3d/symbols/qgspoint3dsymbol.cpp

```diff
@@ -48,6 +74,15 @@ void QgsPoint3DSymbol::writeXml( QDomElement &elem, const QgsReadWriteContext &c
   QDomElement elemTransform = doc.createElement( QStringLiteral( "transform" ) );
   elemTransform.setAttribute( QStringLiteral( "matrix" ), Qgs3DUtils::matrix4x4toString( mTransform ) );
   elem.appendChild( elemTransform );
+
+  if ( billboardSymbol() )
+  {
+    QDomElement symbolElem = QgsSymbolLayerUtils::saveSymbol( QStringLiteral( "symbol" ), billboardSymbol(), doc, context );
+    QString billboardSymbolString;
+    QTextStream stream( &billboardSymbolString );
+    symbolElem.save( stream, -1 );
+    elem.setAttribute( QStringLiteral( "billboard-symbol" ), billboardSymbolString );
```

Comment:
Ummm.... this can be simplified - the returned symbolElem could be simply attached to DOM (without saving the XML to a text stream).